File: algorithms/java/1427.perform-string-shifts.java
class Solution {
    public String stringShift(String s, int[][] shift) {
        StringBuffer ans = new StringBuffer(s);

        int cnt = 0;
        for (int[] arr : shift) {
            if (arr[0] == 0) {
                cnt += arr[1];
            } else {
                cnt -= arr[1];
            }
        }

        int modShift = cnt % s.length();
        System.out.println(modShift);
        if (modShift > 0) {
            ans.append(ans.substring(0, modShift));
            ans.delete(0, modShift);
        } else {
            String sub = ans.substring(ans.length() + modShift, ans.length());
            ans.insert(0, sub);
            ans.delete(ans.length() + modShift, ans.length());
        }
        return ans.toString();
    }
}

File: algorithms/js/1427.perform-string-shifts.js
/**
 * @param {string} s
 * @param {number[][]} shift
 * @return {string}
 */
var stringShift = function(s, shift) {
    let cnt = 0;
    shift.forEach((arr) => {
        if (arr[0] === 0) {
            cnt += arr[1];
        } else {
            cnt -= arr[1];
        }
    });

    let modShift = cnt % s.length;
    if (modShift > 0) {
        s = s.slice(modShift) + s.slice(0, modShift);
    } else {
        let i = s.length + modShift;
        s = s.slice(i) + s.slice(0, i);
    }

    return s;
};
